src/saving/SaveHelper.cs

```diff
@@ -131,11 +131,89 @@ private static Save CreateSaveObject(MainGameState gameState, SaveInformation.Sa
         };
     }
 
+    private static string BuildQuicksaveFilename(uint quicksaveId)
+    {
+        return $"quick_save_{quicksaveId}.{Constants.SAVE_EXTENSION}";
+    }
+
+    /// <summary>
+    ///   Returns the next quicksave-id to be used
+    /// </summary>
+    private static uint GetNextQuicksaveId()
+    {
+        var files = GetSaves();
+        for (uint i = 0; i < Constants.SAVE_MAX_QUICKSAVES_BEFORE_OVERRIDING_OLD_ONES; i++)
+        {
+            if (!files.Contains(BuildQuicksaveFilename(i)))
+            {
+                // Not all 5 quicksaves have been used
+                return i;
+            }
+        }
+
+        // All 5 quicksaves have been used
+        var filename = GetLastSavegame(true);
+        return (uint.Parse(
+                        filename.Substr("quick_save_".Length,
+                        filename.Length - Constants.SAVE_EXTENSION.Length - "quick_save_".Length - 1),
+                        System.Globalization.CultureInfo.InvariantCulture)
+                    + 1) % (Constants.SAVE_MAX_QUICKSAVES_BEFORE_OVERRIDING_OLD_ONES + 1);
```

Comment:
This piece of logic is way too dense and should be split over multiple statements.
Also I'm unsure if it is the best idea to parse strings back to uint. This function could perhaps return a string to make things much cleaner.